src/xwayland/xwm/mod.rs

```diff
@@ -1,18 +1,45 @@
 //!
 //! Xwayland Window Manager module
 //!
-//! Provides an [`X11Wm`] type, which will register itself as a window manager for a previously spawned Xwayland instances,
-//! allowing backwards-compatibility by seemlessly integrating X11 windows into a wayland compositor.
+//! Provides an [`X11Wm`] type, which will register itself as a window manager
+//! for a previously spawned Xwayland instances, allowing
+//! backwards-compatibility by seamlessly integrating X11 windows into a wayland
+//! compositor.
 //!
-//! To use this functionality you must first spawn an [`XWayland`](super::XWayland) instance to attach a [`X11Wm`] to.
+//! To use this functionality you must first spawn an
+//! [`XWayland`](super::XWayland) instance to attach a [`X11Wm`] to.
+//!
+//! # Associating an [X11Surface] with a [WlSurface]
+//!
+//! When an X11 window is created, XWayland will associate it with a
+//! `wl_surface` that it creates via the wayland connection. This happens in two
+//! steps:
+//!
+//!  - Via the [xwayland shell](crate::wayland::xwayland::shell) protocol, a
+//!    serial number is set on the surface, and it is given the
+//!    "xwayland_surface" role.
+//!  - Via the X11 connection, a matching `WL_SURFACE_SERIAL` atom is set on the
+//!    X11 window, which can be queried with
+//!    [`X11Surface::wl_surface_serial()`].
```

Comment:
I don't think this description makes it clear to the user, what they are supposed to do. You describe an internal mechanism from the users perspective. Open questions are:
- Are they supposed to match the surfaces themselves? Or does smithay match them?
- Does this module work without the XwaylandShell protocol?

Reply:
I rewrote this part, let me know what you think.

Reply:
sounds good now!